### PR TITLE
Reconciling static resources

### PR DESCRIPTION
**Pull request by vvondruska**
Restructured the code to allow reconciliation of static resources by watching the default namespace.

## Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] If the change introduces new roles, please consider adding user-friendly descriptions to the roles with the annotation [giantswarm.io/notes](https://github.com/giantswarm/k8smetadata/blob/dff3b2358977e13c90479c66e21c728a96ddfe6d/pkg/annotation/general.go#L12) to help explain their purpose and usage.

### CODE REVIEW

File: service/controller/defaultnamespace/resource/automationsa/create.go [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+			// do nothing
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.Logger().LogCtx(ctx, "level", "info", "message", fmt.Sprintf("serviceaccount %#q in namespace %s has been created", automationSA.Name, namespace))
+
+	} else if err != nil {
+		return microerror.Mask(err)
+	} else {
```
Comment (anvddriesch) [inline service/controller/defaultnamespace/resource/automationsa/create.go:102]: For other resources we use `createOrUpdate` functions so maybe we could have the same for service accounts?

**Review by anvddriesch — APPROVED** [review]
It's such a large PR that I may have skipped through some parts but it all looked very good to me. Just one little thing that caught my attention.
Also let's check that the RBAC documentation we inherited from honeybadger is still up to date :)

File: service/controller/defaultnamespace/resource/automationsa/create.go (hunk 1 above)
Comment (vvondruska) [reply service/controller/defaultnamespace/resource/automationsa/create.go]: I created a `createOrUpdate` function for Service Accounts and put it in its own package.

**Review by mogottsch — APPROVED** [review]
I'm still not too familiar with rbac-operator, but from what I could understand this all looks good

**Pull request merged**